### PR TITLE
cppcheck: some fixes for 4 modules

### PR DESCRIPTION
**Pull request by serval2412**
AnalysisAlgos, AnalysisDataFormats, CalibCalorimetry and CalibFormats

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: CalibCalorimetry/CastorCalib/src/CastorLedAnalysis.cc [hunk 1]
```diff
@@ -441,12 +441,12 @@ void CastorLedAnalysis::LedTrendings(map<HcalDetId, map<int,LEDBUNCH> > &toolT)
 // LED timing - put content and errors
     int j=0;
     for(sample_it=_meol->second[10+m_fitflag].second.first[0].begin();
-        sample_it!=_meol->second[10+m_fitflag].second.first[0].end();sample_it++){
+        sample_it!=_meol->second[+10m_fitflag].second.first[0].end();++sample_it){
```
Comment (davidlt) [inline CalibCalorimetry/CastorCalib/src/CastorLedAnalysis.cc]: This is this is typo? This should stay as `10+m_fitflag`

File: CalibCalorimetry/CastorCalib/src/CastorLedAnalysis.cc [hunk 2]
```diff
@@ -441,12 +441,12 @@ void CastorLedAnalysis::LedTrendings(map<HcalDetId, map<int,LEDBUNCH> > &toolT)
...
       _meol->second[10+m_fitflag].second.second[0]->SetBinContent(++j,*sample_it);
     }
     j=0;
     for(sample_it=_meol->second[10+m_fitflag].second.first[1].begin();
-        sample_it!=_meol->second[10+m_fitflag].second.first[1].end();sample_it++){
+        sample_it!=_meol->second[+10m_fitflag].second.first[1].end();++sample_it){
```
Comment (davidlt) [inline CalibCalorimetry/CastorCalib/src/CastorLedAnalysis.cc]: Same here, `10+m_fitflag` should not be modified.

**Comment by serval2412** [comment]
Argh, I mistake with sed command, sorry for the noise.

**Pull request closed without merging**